dbms/src/Flash/Coprocessor/DAGContext.cpp

```diff
@@ -173,14 +179,19 @@ void DAGContext::initOutputInfo()
     keep_session_timezone_info = encode_type == tipb::EncodeType::TypeChunk || encode_type == tipb::EncodeType::TypeCHBlock;
 }
 
-String DAGContext::getRootExecutorId()
+void DAGContext::initListBasedExecutors()
```

Comment:
rename to `initExecutorDAG` and then check `executor.has_executor_id` for tree base executor here?